src/utils/retry_utils/src/retry_utils.c

```diff
@@ -62,3 +62,116 @@ time_t ADUC_Retry_Delay_Calculator(
         ADUC_GetTimeSinceEpochInSeconds() + (time_t)additionalDelaySecs + (time_t)(delay * (1 + jitterPercent)));
     return retryTimestampSec;
 }
+
+void ADUC_Retriable_Operation_Init(ADUC_Retriable_Operation_Context* context, bool startNow)
+{
+    if (context == NULL)
+    {
+        return;
+    }
+
+    context->state = ADUC_Retriable_Operation_State_NotStarted;
+    context->nextExecutionTime = 0;
+    context->expirationTime = 0;
+    context->attemptCount = 0;
+
+    if (startNow)
+    {
+        context->nextExecutionTime = ADUC_GetTimeSinceEpochInSeconds();
+    }
+}
+
+/**
+ * @brief Perform a retriable operation.
+ */
+bool ADUC_Retriable_Operation_DoWork(ADUC_Retriable_Operation_Context* context)
```

Comment:
Add unit tests to verify robustness and that state machine transitions as intended